fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -496,7 +496,11 @@ public ParseNode visitCreateTableStatement(StarRocksParser.CreateTableStatementC
                 extProperties.put(property.getKey(), property.getValue());
             }
         }
-
+        if (context.orderByDesc() != null) {
+            if (context.keyDesc() == null || getKeysDesc(context.keyDesc()).getKeysType() != KeysType.PRIMARY_KEYS) {
+                throw new IllegalArgumentException("only primary key support sort key");
```

Comment:
This logic should be checked in CreateTableAnalyzer, ASTBuilder is only responsible for syntax error checking

Reply:
changed